app/models/manageiq/providers/ibm_cloud/object_storage/storage_manager.rb

```diff
@@ -0,0 +1,211 @@
+                      },
+                      {
+                        :component  => "text-field",
+                        :name       => "authentications.default.auth_key",
+                        :id         => "authentications.default.auth_key",
+                        :label      => _("Apikey"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
+                        :type       => "password",
+                      },
+                      {
+                        :component  => "text-field",
+                        :name       => "authentications.bearer.userid",
+                        :id         => "authentications.bearer.userid",
+                        :label      => _("Access Key"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
+                      },
+                      {
+                        :component  => "text-field",
+                        :name       => "authentications.bearer.password",
+                        :id         => "authentications.bearer.password",
+                        :label      => _("Secret Key"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
+                        :type       => "password",
+                      }
+                    ]
+                  }
+                ]
+              }
+            ]
+          ]
+        }
+      ]
+    }.freeze
+  end
+
+  def self.hostname_required?
+    false
+  end
+
+  def self.supported_for_create?
+    true
+  end
+
+  def self.ems_type
+    @ems_type ||= "ibm_cloud_object_storage".freeze
+  end
+
+  def self.description
+    @description ||= "IBM Cloud Object Storage".freeze
+  end
+
+  def image_name
+    "ibm"
+  end
+
+  def cos_creds
+    iam     = authentications.detect { |e| e.authtype == 'default' }
+    bearer  = authentications.detect { |e| e.authtype == 'bearer' }
+    endp    = endpoints.first
+
+    guid   = uid_ems
+    region = provider_region
+    endpoint = endp.url
+    apikey   = iam.auth_key
+    access_key = bearer.userid
+    secret_encr = bearer.password
+    secret_key  = ManageIQ::Password.try_decrypt(secret_encr)
+
+    return guid, apikey, region, endpoint, access_key, secret_key
+  end
+
+  def connect(_args = {})
+    _, _, region, endpoint, access_key, secret_key = cos_creds
+    self.class.raw_connect(region, endpoint, access_key, secret_key)
+  end
+
+  def verify_credentials(_args = {})
+    guid, apikey, region, endpoint, access_key, secret_key = cos_creds
+    self.class.verify_bearer(region, endpoint, access_key, secret_key)
+    self.class.verify_iam(guid, apikey)
+
+    true
+  end
+
+  def self.verify_credentials(args)
+    guid     = args["uid_ems"]
+    auth_key = args.dig("authentications", "default", "auth_key")
+    apikey   = ManageIQ::Password.try_decrypt(auth_key)
```

Comment:
We should make sure to handle the case where we are editing an existing provider (aka an "id" will be passed and none of the password fields will be present)

See https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/vpc/manager_mixin.rb#L87

Reply:
I see. If I get the example right then it looks for the creds in the args and if not found then looks for the already existing/saved ones.

Reply:
:+1: yes exactly